### ISSUE TITLE
Tailwind's classes do not work in html elements (SvelteKit v1.0.0-next.102)

### ISSUE THREAD

**Issue by pablohgm** [svelte-add/tailwindcss#42]
Hi, I was trying to get a clean installation of svelte-kit and tailwind
I did
```
npm init svelte@next appName
npx svelte-add tailwindcss
npm i
```

The installation is ok, `@apply` styles works as expected, but when I tried to use tailwind's classes in the elements, the styles do not work, for example:
`<p class='text-red-50'>test</p>`

FYI
```
✔ Which Svelte app template? › R/ SvelteKit demo app
✔ Use TypeScript? … R/ No
✔ Add ESLint for code linting? … R/ Yes
✔ Add Prettier for code formatting? … R/ Yes
```

**Comment by babichjacob** [comment]
This is a bug with this project attributed to the recent `$layout.svelte` -> `__layout.svelte` change in SvelteKit. Hold tight while I fix this.

**Comment by dit7ya** [comment]
Is there a workaround for this in the meantime?

**Comment by babichjacob** [comment]
> Is there a workaround for this in the meantime?

Apply the current version of this preset with `pnpx svelte-add tailwindcss --jit`.

Create (or edit) a `src/routes/__layout.svelte` with these contents:

```svelte
<script>
  import "../app.postcss";
</script>

<slot />
```

and an `src/app.postcss` with these contents:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

I, too, am baffled by why this isn't working as expected right now (without manual intervention). Slating this as something to fix with the rewrite and giving up on the existing preset because it's somehow broken beyond repair.

**Comment by jormaj** [comment]
I've found that there's a difference between the following two divs:
```
<div
	class="shadow-md p-4 border-2"
	class:border-green-600={true}
	class:shadow-lg={true}
	class:bg-green-600={true}
>
```
and
```
<!-- prettier-ignore -->
<div class="shadow-md p-4 border-2" class:border-green-600={true} class:shadow-lg={true} class:bg-green-600={true}>
```
In the first case only the classes on the first line get applied (shadow-md, p-4, border-2), in the second case all classes get applied.
So maybe a regular expression-like bug causing a hickup when classes are spread over multiple lines..?

**Comment by brunetton** [comment]
> ```css
> ```css
> @tailwind base;
> @tailwind components;
> @tailwind utilities;
> ```
> 
> 
>       
> ```

Thanks @unknownterritory; I can confirm that this fixes the problem for me :+1:

**Comment by PotatoKilla** [comment]
> It seems to work with the default
> 
> ```shell
> npm init svelte@next appName
> npx svelte-add tailwindcss
> npm i
> ```
> 
> followed by adding to `app.css`
> 
> ```css
> @tailwind base;
> @tailwind components;
> @tailwind utilities;
> ```

@unknownterritory im a bit new so im sorry if this is a dumb Q but i see you said "add to app.css" do i have to make a file named that and place it in the src folder? or am i missing something. thanks

**Comment by babichjacob** [comment]
> @unknownterritory im a bit new so im sorry if this is a dumb Q but i see you said "add to app.css" do i have to make a file named that and place it in the src folder? or am i missing something. thanks

It should be the case that either `src/app.css` or `src/app.postcss` exists for you to place that in after running this program's command.

**Comment by PotatoKilla** [comment]
> > @unknownterritory im a bit new so im sorry if this is a dumb Q but i see you said "add to app.css" do i have to make a file named that and place it in the src folder? or am i missing something. thanks
> 
> It should be the case that either `src/app.css` or `src/app.postcss` exists for you to place that in after running this program's command.
@babichjacob 

this is what i have
![helpCapture](https://user-images.githubusercontent.com/77354712/118379637-ba9ec580-b5a9-11eb-9608-68771a8520aa.PNG)

**Comment by babichjacob** [comment]
> this is what i have
> ![helpCapture](https://user-images.githubusercontent.com/77354712/118379637-ba9ec580-b5a9-11eb-9608-68771a8520aa.PNG)

Looks like you'll have to make a new file then and make sure it's imported in `src/routes/__layout.svelte` like so:

```svelte

<script>
  import "../app.postcss";
</script>

<slot />
```

Reminder that this is a bug and that all of this was supposed to have been taken care of for you 😔.

**Comment by PotatoKilla** [comment]
@babichjacob yeah i was bangin my head on the desk for a bit till i realized other people was having the same issue lol im so new i never know when im just being dumb or if its actually not working lol

ok so im gonna ask you to dumb it down for me so i dont mess up lol
what should the file be named and what should be the contents of that file

i really appreciate the help dude

**Comment by PotatoKilla** [comment]
@babichjacob fixed, thanks alot!

**Comment by DrewRidley** [comment]
For some reason I have noticed that although this issue appears to be fixed, it might not be. Adding custom colors to your tailwind config immediately breaks roughly half the styles in tailwind, and worse yet, even adding
```
colors: {}
``` 
still breaks it! Very strange bug worth investigating!

**Comment by babichjacob** [comment]
> For some reason I have noticed that although this issue appears to be fixed, it might not be. Adding custom colors to your tailwind config immediately breaks roughly half the styles in tailwind, and worse yet, even adding
> 
> ```
> colors: {}
> ```
> 
> still breaks it! Very strange bug worth investigating!

There's a very small chance this is not within my power. Either way, what is your full Tailwind config?

**Comment by DrewRidley** [comment]
The tailwind config is pretty default, nothing changed except adding colors.

```
const { tailwindExtractor } = require("tailwindcss/lib/lib/purgeUnusedStyles");

module.exports = {
	mode: "jit",
	purge: {
		content: [
			"./src/**/*.{html,js,svelte,ts}",
		],
		options: {
			defaultExtractor: (content) => [
				// If this stops working, please open an issue at https://github.com/svelte-add/tailwindcss/issues rather than bothering Tailwind Labs about it
				...tailwindExtractor(content),
				// Match Svelte class: directives (https://github.com/tailwindlabs/tailwindcss/discussions/1731)
				...[...content.matchAll(/(?:class:)*([\w\d-/:%.]+)/gm)].map(([_match, group, ..._rest]) => group),
			],
		},
		safelist: [/^svelte-[\d\w]+$/],
	},
	theme: {
		colors: {

		},
		extend: {},
	},
	variants: {
		extend: {},
	},
	plugins: [],
};
```
that should break it. If not, you can try changing colors to

```
		colors: {
			x: {
                DEFAULT: '#323448'
            }
		},
``` 
and I can guarantee it to be reproduceable.

**Comment by babichjacob** [comment]
Your colors are overriding the defaults rather than being added to them: https://tailwindcss.com/docs/customizing-colors#extending-the-defaults

```js
const { tailwindExtractor } = require("tailwindcss/lib/lib/purgeUnusedStyles");

module.exports = {
	mode: "jit",
	purge: {
		content: [
			"./src/**/*.{html,js,svelte,ts}",
		],
		options: {
			defaultExtractor: (content) => [
				// If this stops working, please open an issue at https://github.com/svelte-add/tailwindcss/issues rather than bothering Tailwind Labs about it
				...tailwindExtractor(content),
				// Match Svelte class: directives (https://github.com/tailwindlabs/tailwindcss/discussions/1731)
				...[...content.matchAll(/(?:class:)*([\w\d-/:%.]+)/gm)].map(([_match, group, ..._rest]) => group),
			],
		},
		safelist: [/^svelte-[\d\w]+$/],
	},
	theme: {
		colors: {

		},
		extend: {},
	},
	variants: {
		extend: {},
	},
	plugins: [],
};
const { tailwindExtractor } = require("tailwindcss/lib/lib/purgeUnusedStyles");

module.exports = {
	mode: "jit",
	purge: {
		content: [
			"./src/**/*.{html,js,svelte,ts}",
		],
		options: {
			defaultExtractor: (content) => [
				// If this stops working, please open an issue at https://github.com/svelte-add/tailwindcss/issues rather than bothering Tailwind Labs about it
				...tailwindExtractor(content),
				// Match Svelte class: directives (https://github.com/tailwindlabs/tailwindcss/discussions/1731)
				...[...content.matchAll(/(?:class:)*([\w\d-/:%.]+)/gm)].map(([_match, group, ..._rest]) => group),
			],
		},
		safelist: [/^svelte-[\d\w]+$/],
	},
	theme: {
		extend: {
			colors: {

			},
		},
	},
	variants: {
		extend: {},
	},
	plugins: [],
};
```

**Comment by DrewRidley** [comment]
Ohhhh!!! that explains a LOT. Thanks so much for the help. I did not read the docs correctly.

**Comment by erlendtryti** [comment]
Is there an ETA on this fix?

**Comment by babichjacob** [comment]
> Is there an ETA on this fix?

Not quite. I have some local / unpushed and uncommitted changes as I've been encountering obstacles.

**Comment by babichjacob** [comment]
🎉 This has been fixed! 🎉

Due to the current state of `svelte-add`, you may encounter a different issue when using this tool. Please feel free to find an existing issue at https://github.com/svelte-add/svelte-add/issues or open a new one for whatever problem you encounter.

Thank you everyone for your patience and continued interest in the project!
